qiskit_dynamics/pulse_simulator/pulse_simulator.py

```diff
@@ -0,0 +1,526 @@
+
+from qiskit_dynamics.solvers.solver_classes import Solver
+
+from .dynamics_job import DynamicsJob
+from .pulse_simulator_utils import (
+    _get_dressed_state_decomposition,
+    _get_lab_frame_static_hamiltonian,
+    _get_memory_slot_probabilities,
+    _sample_probability_dict,
+    _get_counts_from_samples,
+)
+
+
+class PulseSimulator(BackendV2):
+    r"""Pulse enabled simulator backend.
+
+    **Supported options**
+
+    * ``shots``: Number of shots per experiment. Defaults to ``1024``.
+    * ``solver``: The Qiskit Dynamics :class:`.Solver` instance used for simulation.
+    * ``solver_options``: Dictionary containing optional kwargs for passing to :meth:`Solver.solve`,
+      indicating solver methods and options. Defaults to the empty dictionary ``{}``.
+    * ``subsystem_dims``: Dimensions of subsystems making up the system in ``solver``. Defaults
+      to ``[solver.model.dim]``.
+    * ``subsystem_labels``: Integer labels for subsystems. Defaults to
+      ``[0, ..., len(subsystem_dims) - 1]``.
+    * ``meas_map``: Measurement map. Defaults to ``[[idx] for idx in subsystem_labels]``.
+    * ``initial_state``: Initial state for simulation, either the string ``"ground_state"``,
+      indicating that the ground state for the system Hamiltonian should be used, or an arbitrary
+      ``Statevector`` or ``DensityMatrix``. Defaults to ``"ground_state"``.
+    * ``normalize_states``: Boolean indicating whether to normalize states before computing
+      outcome probabilities. Defaults to ``True``. Setting to ``False`` can result in errors if
+      the solution tolerance results in probabilities with significant numerical deviation from
+      proper probability distributions.
+    * ``meas_level``: Form of measurement return. Only supported value is ``2``, indicating that
+      counts should be returned. Defaults to ``meas_level==2``.
+    * ``max_outcome_level``: For ``meas_level==2``, the maximum outcome for each subsystem.
+      Values will be rounded down to be no larger than ``max_outcome_level``. Must be a positive
+      integer or ``None``. If ``None``, no rounding occurs. Defaults to ``1``.
+    * ``memory``: Boolean indicating whether to return a list of explicit measurement outcomes for
+      every experimental shot. Defaults to ``True``.
+    * ``seed_simulator``: Seed to use in random sampling. Defaults to ``None``.
+    * ``experiment_result_function``: Function for computing the ``ExperimentResult``
+      for each simulated experiment. This option defaults to
+      :func:`default_experiment_result_function`, and any other function set to this option
+      must have the same signature. Note that the default utilizes various other options that
+      control results computation, and hence changing it will impact the meaning of other options.
+    """
+
+    def __init__(
+        self,
+        solver: Solver,
+        target: Optional[Target] = None,
+        provider: Optional[Provider] = None,
```

Comment:
Is the provider kwarg actually needed? We dont have one in dynamics, so maybe can just leave as None for now when calling the super init

Reply:
Removed it. Removed it from the `super().__init__` call as well as it defaults to `None` anyway.